modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheMetricsImpl.java

```diff
@@ -513,6 +526,61 @@ public void delegate(CacheMetricsImpl delegate) {
         return cacheSize.value();
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public long getCacheTouchHits() {
+        return cacheTouchHits.value();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public long getCacheTouchMisses() {
+        return cacheTouchMisses.value();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public long getCacheTouches() {
+        return cacheTouches.value();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public float getCacheTouchHitPercentage() {
+        long hits = cacheTouchHits.value();
+        long touches = cacheTouches.value();
+
+        if (touches == 0)
+            return 0;
+
+        return (float) hits / touches * 100.0f;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public float getCacheTouchMissPercentage() {
+        long misses = cacheTouchMisses.value();
+        long touches = cacheTouches.value();
+
+        if (touches == 0)
+            return 0;
+
+        return (float) misses / touches * 100.0f;
+    }
+
+    public void onCacheTouch() {
+        cacheTouches.increment();
+    }
+
+    public void onCacheTouchHit() {
```

Comment:
I could not find any usage of this method except tests. Perhaps, some code changes are not included in the pull request.

Reply:
Hi @sk0x50 
Thank you for pointing out the missing invocation of the metrics. Previously, I had integrated them in `GridCacheAdapter.java`, and it seems they were missed in this PR.

Upon further investigation, I looked at how similar metrics, such as getCachePuts(), are handled and noticed that classes like `CacheMetricsSnapshot.java` and `VisorCacheMetrics.java` track similar cache operations and metrics. 
<img width="834" alt="Screenshot 2024-09-16 at 6 14 29 AM" src="https://github.com/user-attachments/assets/321f8047-2ca4-4dda-a85e-60cb7b00b4e6">
These appear to be better candidates for invoking the cache touch metrics (`onCacheTouch()`, `onCacheTouchHit()`, `onCacheTouchMiss()`).

Could you kindly confirm that these classes are appropriate for integrating the touch metrics?

Also, I wanted to clarify a point regarding `CacheMetricsSnapshot` and `CacheMetricsSnapshotV2`: should both files be updated, or would just V2 alone be sufficient?

Thanks.